Yoast/javascript
# Add passive voice assessment for Indonesian

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds passive voice assessment for Indonesian, including a check, a number or specs and a list of exceptions. Two regexes used for the recognition of the past participle in Dutch were also modified to accommodate a broader range of punctuation marks.

* [Yoast Seo Free] Adds an assessment for the recognition of the passive voice in Indonesian texts (for example: "Berita itu sudah didengar oleh Anwar kemarin") making sure that passive infinitives are treated as active sentences (for example: "Kalimat yang lebih baik untuk digunakan").

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Run `yarn test` and make sure all tests still pass.

* In the content-analysis app:
  * Set the locale to id_ID;
  * Paste [this text](https://docs.google.com/document/d/1oShcLuQLscN1WCLbjDk_jLOqgQvJEw24wz-aCmZMsV4/edit) in the Text input window (it won't detect any passives);
  * Add these sentences containing Indonesian passive verb forms: `Penumpang itu dilempar ke luar`, 
`Sudahkah anak merasa didengarkan?`, `Suratnya belum dikirimkan ke alamat yang benar`. The passive voice assessment should read `11.2% of the sentences contain passive voice, which is more than the recommended maximum of 10%. Try to use their active counterparts.`;
  * Add `untuk` before every passive verb form. The assessment bullet should turn green and read `You're using enough active voice. That's great!`;
  * Add words from the many exceptions to passives (e.g.: `diskontinuitas, diskualifikasi, diskriminatif, diskriminator, digitalisasi,
disinformasi`). The passive voice assessment should not change.
  * Delete all the occurrences of `untuk` from the text. The passive voice assessment should read `11.2% of the sentences contain passive voice, which is more than the recommended maximum of 10%. Try to use their active counterparts.`

## QA Testing Instructions
 * Set the wp language to Bahasa Indonesia
 * You should see a bullet assessment for passive verb forms

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * Detection of passive voice in Dutch, where the past participle form is followed by punctuation marks, such as: `!?,.")>']`


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-481
